user_tools/src/spark_rapids_pytools/rapids/qualification.py

```diff
@@ -824,16 +825,20 @@ def process_df_for_stdout(raw_df):
         if not self._evaluate_rapids_jar_tool_output_exist():
             return
         # process the output through the XGboost model if enabled
+        predictions_df = None
         if self.ctxt.get_ctxt('estimationModel') == QualEstimationModel.XGBOOST:
-            xgboost_input_dir = self.ctxt.get_local('outputFolder')
-            xgboost_output_dir = xgboost_input_dir
-            predict(qual=xgboost_input_dir, profile=xgboost_input_dir, output_dir=xgboost_output_dir)
+            model_name = self.ctxt.platform.get_prediction_model_name()
+            input_dir = self.ctxt.get_local('outputFolder')
+            output_info = self.__build_prediction_output_files_info()
+            predictions_df = predict(platform=model_name, qual=input_dir,
+                                     profile=input_dir, output_info=output_info)
 
         rapids_output_dir = self.ctxt.get_rapids_output_folder()
         rapids_summary_file = FSUtil.build_path(rapids_output_dir,
                                                 self.ctxt.get_value('toolOutput', 'csv', 'summaryReport', 'fileName'))
         self.ctxt.logger.debug('Rapids CSV summary file is located as: %s', rapids_summary_file)
         df = pd.read_csv(rapids_summary_file)
+        df = self.__update_apps_with_prediction_info(df, predictions_df)
```

Comment:
Our prediction code is not stable yet and it will crash the Tools if it runs on an older Jar.
For now, to avoid a total failure lets have this inside try-catch block.
If the prediction fails because of schema error or incorrect input arguments, we can recover by just showing the exception and using the original df.

Later, we can handle each kind of error on its own inside the prediction code.

Reply:
Thanks @amahussein. Added try-catch block.